src/core/crypto/ed25519.ts

```diff
@@ -147,17 +147,30 @@ export class Ed25519PrivateKey extends Serializable implements PrivateKey {
    * Create a new PrivateKey instance from a Uint8Array or String.
    *
    * @param hexInput HexInput (string or Uint8Array)
+   * @param strict If set, enforces private key has to use AIP-80
    */
-  constructor(hexInput: HexInput) {
+  constructor(hexInput: HexInput, strict: boolean = false) {
     super();
-
-    const privateKeyHex = Hex.fromHexInput(hexInput);
-    if (privateKeyHex.toUint8Array().length !== Ed25519PrivateKey.LENGTH) {
+    let data: Hex;
+    if (typeof hexInput === "string") {
+      // A non-prefixed input
+      if (!strict && hexInput.startsWith("0x")) {
```

Comment:
`fromHexInput` uses `fromHexString` - do we really need this check?

Reply:
Need it to tell that it's not prefixed